File: TabbarCustom/TabbarCustom/ViewController.m
//
//  ViewController.m
//  TabbarCustom
//
//  Created by lk on 16/4/24.
//  Copyright © 2016年 LK. All rights reserved.
//

#import "ViewController.h"
#import "PDTabbarViewController.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}
- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    UIViewController *view4 = [[UIViewController alloc] init];
    view4.view.backgroundColor = [UIColor redColor];
    
    [(PDTabbarViewController *)self.tabBarController setMTabBarHidden:YES];
    
    view4.hidesBottomBarWhenPushed= YES;
    
    [self.navigationController pushViewController:view4 animated:YES];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 事件的传递和响应机制/事件的传递和响应机制/ViewController.swift
//
//  ViewController.swift
//  事件的传递和响应机制
//
//  Created by lk on 16/7/14.
//  Copyright © 2016年 LK. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    
    //http://www.jianshu.com/p/2e074db792ba
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let whiteView = WhiteView(frame:CGRectMake(100, 100, 300, 300))
//        whiteView.frame = CGRectMake(100, 100, 300, 300)
        whiteView.backgroundColor = UIColor.grayColor()
        view.addSubview(whiteView)
        
        
        let redView = RedView()
        redView.frame = CGRectMake(100, 100, 100, 100)
        redView.backgroundColor = UIColor.redColor()
        whiteView.addSubview(redView)
        
        let greenView = GreenView()
        greenView.frame = CGRectMake(100, 200, 100, 100)
        greenView.backgroundColor = UIColor.greenColor()
        whiteView.addSubview(greenView)
        
        
     
    }

    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        print("这是父控件")
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }


}

